daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/ValueCoder.scala

```diff
@@ -427,7 +428,10 @@ object ValueCoder {
           case ValueInt64(i) =>
             builder.setInt64(i).build()
           case ValueNumeric(d) =>
-            builder.setDecimal(Numeric.toUnscaledString(d)).build()
+            if (useLegacyDecimal(valueVersion))
```

Comment:
Minor: you may want to evaluate using a guard on the pattern matching rather than nesting an `if` (doesn't change much apart from less nesting, purely a stylistic preference, ignore at will).